scripts/db/src/login.ts

```diff
@@ -14,8 +17,23 @@ type PersonResult = {
   last_name: string
   password: string
 }
+/** Forum user */
+type ForumUser = {
+  _id: string
+  displayName: string
+  services: {
+    password: {
+      bcrypt: string
+    }
+  }
+  /**
+   * Email address is also stored in an email field, but as far as
+   * authentication is concerned, this is how we store emails
```

Comment:
I wonder if there's a good argument to be made that this should use the `email` field. Presumably this is for consistency with the current Forum implementation, which I'm guessing allows logins using a secondary email.

This maybe raises a question of whether we should figure out how to harmonise multiple emails across the two DBs. Like, if I try to log in with a secondary email on the Forum, and my account is created with that address, but it's not the same as the email in use on Parfit, then we'll end up with a separate account for each service...

Thinking about it a second longer, I think this is just the way it is at this part of the process. Still if people want to merge their accounts, we should think about how to do that as painlessly as possible.

Reply:
Related question – how are we planning to manage email changes. For example, at the moment if I change an email on the parfit side, I sync that back to Auth0. If the user signs into the Forum with that new email on the token, then I guess you'll reconcile it on your side by updating Mongo, matching on the Auth0 user ID. I'm guessing that's just fine, but I'm wondering if we need to consider any weird out-of-sync situations here...

Reply:
For syncing email changes, I've recorded a todo.

Reply:
Note that it should be very rare for a user to actually have multiple email addresses. I think an admin needs to have manually done that.

So then we just collapse to the question of what to do with multiple emails across services  that use login.EA.org. I think those should be handled by an easy merge process. ...More work, wooo. But not this PR.